### DISCUSSION TITLE
Bard output not converted to a Statamic Sets/collection.

### DISCUSSION THREAD

**Discussion by ghost** [statamic-rad-pack/runway#634] [Help] [answered]
### Description

HI! I am working on integrating Runway in a Statamic application within a specific (Laravel) Blade view, returned by a regular controller.

The field in question, pdp_content, is a Bard field stored as JSON. When dumped into the Blade view, the data is converted from JSON into a PHP array using the model's casts attribute:

```
'pdp_content' => 'json',
```

However, I need the data structured in a way that mimics a regular Bard field's sets. This would allow me to reuse existing field sets without altering the current HTML structure.

## Steps to reproduce

1. Create a Runway resource with a Bard field (e.g., pdp_content).
2. Set up a Laravel model with the following casts attribute:

```
'pdp_content' => 'json',
```
3. Pass the Runway resource's pdp_content field to a Laravel Blade view.
4. Observe the output as a plain array instead of being structured as Bard sets.

## What i tried

1. Using the [Bard Items modifier](https://statamic.dev/modifiers/bard_items), but it doesn’t seem to apply in this context.
2. Implementing a custom Cast class to transform the JSON data into a Statamic Value class within the model.

## Expected Behaviour

The array output of the Bard field (pdp_content) should be structured as a Bard set collection, matching the behavior of Statamic's native Bard fields. 

## Question

Is there a way to convert a regular PHP array (from the pdp_content field) into a Bard set collection? This would allow me to seamlessly integrate with existing Bard field sets in Statamic without modifying the underlying HTML.

This is for a project where I’m leveraging Statamic Runway and integrating it into a Laravel Blade view. The goal is to maintain consistency with existing Bard field data structures.

Thank you for your help!

### Environment

Environment
Application Name: Duurzaaminvesteren.nl
Laravel Version: 11.34.1
PHP Version: 8.3.14
Composer Version: 2.7.7
Environment: local
Debug Mode: ENABLED
URL: www.duurzaaminvesteren.test
Maintenance Mode: OFF
Timezone: Europe/Amsterdam
Locale: nl

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: reverb
Cache: file
Database: mariadb
Logs: stack / single, flare
Mail: smtp
Queue: sync
Session: database

Locales
Installed: en, nl
LaravelLang\Attributes\Plugin: 2.11.1
LaravelLang\Lang\Plugin: 15.9.5
Locales Version: 2.9.2
Protected: en, nl
Publisher Version: 16.4.0

Filament
Blade Icons: NOT CACHED
Packages: filament, forms, notifications, support, tables
Panel Components: NOT CACHED
Version: v3.2.124
Views: NOT PUBLISHED

Livewire
Livewire: v3.5.12

Statamic
Addons: 3
Sites: 1
Stache Watcher: Enabled (auto)
Static Caching: Disabled
Version: 5.40.0 PRO

Statamic Addons
rias/statamic-redirect: 3.8.2
statamic-rad-pack/runway: 7.12.0
statamic/seo-pro: 6.3.0

**Comment by duncanmcclean** [comment]
> Is there a way to convert a regular PHP array (from the pdp_content field) into a Bard set collection? This would allow me to seamlessly integrate with existing Bard field sets in Statamic without modifying the underlying HTML.

Hmm, there's isn't anything in Statamic or Runway for that.

You'd need to write a command that goes through your array, maps them to the Bard structure, then updates the `pdp_content` column on your model.

Alternatively, you could *maybe* write some kind of accessor/mutator to handle it.

**Reply by ghost** [reply]
Yeah, as a first step im updating the array with a custom Cast class

```
public function get($model, string $key, $value, array $attributes)
    {

        if (is_null($value)) {
            return null;
        }

        $fieldtype = new Bard();
        return new Value(
            json_decode($value, true),
            $key,
            $fieldtype,
            $model
        );
    }
```
but this is braking the CP because most likely im forgetting something.

**Reply by duncanmcclean** [reply]
You'll need to transform the array to the structure that Bard expects. Passing your non-Bard array to Bard won't work.

**Reply by ghost** [reply, marked as answer]
Hi Duncan,

I've looked into this with my colleague. He found a solution by iterating over the data like this:

```
        @foreach($proposition->newAugmentedInstance()->get('pdp_content') as $item)
            @includeIf('components.organisms.blocks.'.$item->type, ['content' => $item])
        @endforeach
```   
and casting the data to an array via the model.
